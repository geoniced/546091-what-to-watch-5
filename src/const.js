export const ALL_GENRES_FILTER = `All genres`;
