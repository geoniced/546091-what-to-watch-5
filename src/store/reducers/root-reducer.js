import {combineReducers} from "redux";
import {filmsData} from "./films-data/films-data";

export const NameSpace = {
  DATA: `DATA`,
};

export default combineReducers({
  [NameSpace.DATA]: filmsData,
});
