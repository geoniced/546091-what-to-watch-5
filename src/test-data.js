export const noop = () => {};

export const movieCard = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: 2014,
};

export const filmListMock = [
  {
    backgroundColor: `#73B39A`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    rating: 3.6,
    ratingsCount: 109661,
    releaseYear: 2008,
    runtime: 92,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    title: `Bronson`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 2,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    rating: 1.5,
    ratingsCount: 1635992,
    releaseYear: 1994,
    runtime: 153,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    title: `Pulp Fiction`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#828585`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg`,
    description: `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
    director: `Jeff Nichols`,
    genre: `Action`,
    id: 3,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg`,
    rating: 3.3,
    ratingsCount: 67815,
    releaseYear: 2016,
    runtime: 112,
    starring: [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst `],
    title: `Midnight Special`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#B9B27E`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    description: `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
    director: `Wachowski Brothers`,
    genre: `Action`,
    id: 4,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    rating: 4.4,
    ratingsCount: 1503092,
    releaseYear: 1999,
    runtime: 136,
    starring: [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
    title: `Matrix`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#E1DAD7`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
    description: `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
    director: `Brian Helgeland`,
    genre: `Crime`,
    id: 5,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
    rating: 3.5,
    ratingsCount: 138487,
    releaseYear: 2015,
    runtime: 132,
    starring: [`Tom Hardy`, `Emily Browning`, `Taron Egerton`],
    title: `Legend`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#A6B7AC`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/gangs_of_new_york.jpg`,
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    director: `Martin Scorsese`,
    genre: `Crime`,
    id: 6,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    rating: 8.8,
    ratingsCount: 370881,
    releaseYear: 2002,
    runtime: 167,
    starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
    title: `Gangs of new york`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#C4C0C0`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/A_Star_is_Born.jpg`,
    description: `A musician helps a young singer find fame as age and alcoholism send his own career into a downward spiral.`,
    director: `Bradley Cooper`,
    genre: `Drama`,
    id: 7,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_Is_Born.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg`,
    rating: 3.9,
    ratingsCount: 244484,
    releaseYear: 2018,
    runtime: 136,
    starring: [`Lady Gaga`, `Bradley Cooper`, `Sam Elliott`],
    title: `A Star Is Born`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#EBC996`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/beach.jpg`,
    description: `Vicenarian Richard travels to Thailand and finds himself in possession of a strange map. Rumours state that it leads to a solitary beach paradise, a tropical bliss. Excited and intrigued, he sets out to find it.`,
    director: `Danny Boyle`,
    genre: `Adventure`,
    id: 8,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/beach.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg`,
    rating: 3.3,
    ratingsCount: 207824,
    releaseYear: 2000,
    runtime: 119,
    starring: [`Leonardo DiCaprio`, `Daniel York`, `Patcharawan Patarakijjanon`],
    title: `Beach`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#D6CDAF`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Aviator.jpg`,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    director: `Martin Scorsese`,
    genre: `Drama`,
    id: 9,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Aviator.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg`,
    rating: 9.8,
    ratingsCount: 307174,
    releaseYear: 2014,
    runtime: 170,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    title: `Aviator`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#F0DBA2`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Johnny_English.jpg`,
    description: `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
    director: `Peter Howitt`,
    genre: `Comedy`,
    id: 10,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Johnny_English.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg`,
    rating: 10,
    ratingsCount: 136843,
    releaseYear: 2003,
    runtime: 88,
    starring: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
    title: `Johnny English`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  }
];
