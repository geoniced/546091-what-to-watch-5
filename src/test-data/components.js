import React from "react";
import PropTypes from "prop-types";

const MockComponentWithChildren = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponentWithChildren.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export {MockComponentWithChildren};


const MockComponent = () => <div />;
export {MockComponent};
