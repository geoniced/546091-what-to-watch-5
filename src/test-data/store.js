import {NameSpace} from "../store/reducers/root-reducer";
import {AuthorizationStatus} from "../const";
import configureStore from "redux-mock-store";
import {filmListMock} from "./test-data";

const configuredStore = configureStore([]);
const mockedStore = configuredStore({
  [NameSpace.DATA]: {
    activeGenre: `All genres`,
    films: filmListMock,
    shownFilmsCount: 8,
    isLoading: false,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
  }
});

export {mockedStore};
