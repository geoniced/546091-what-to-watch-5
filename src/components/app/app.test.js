import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {filmListMock, movieCard} from "../../test-data";
import App, {App as AppWithoutStore} from "./app";
import {noop} from "../../test-data";
import {NameSpace} from "../../store/reducers/root-reducer";
import {AuthorizationStatus} from "../../const";

describe(`App render`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let appComponent = null;

  // beforeEach(() => {
  store = mockStore({
    [NameSpace.DATA]: {
      activeGenre: `All genres`,
      films: filmListMock,
      shownFilmsCount: 8,
      isLoading: false,
    },
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
    }
  });
  // });

  store.dispatch = jest.fn();

  appComponent = renderer.create(
      <Provider store={store}>
        <App
          movieCard={movieCard}
          isLoading={false}
          films={[]}
          loadFilmList={noop}
        />
      </Provider>,
      {
        createNodeMock: () => {
          return {};
        }
      }
  );

  it(`renders App component`, () => {

    expect(appComponent.toJSON()).toMatchSnapshot();
  });

  it(`renders App component's preloader (no store here)`, () => {
    const tree = renderer
      .create(
          <AppWithoutStore
            movieCard={movieCard}
            films={[]}
            isLoading={true}
            loadFilmList={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
