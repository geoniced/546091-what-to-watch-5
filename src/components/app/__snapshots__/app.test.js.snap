// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render renders App component 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link "
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
          data-genre="All genres"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          data-genre="Action"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Action
          </a>
        </li>
        <li
          className="catalog__genres-item "
          data-genre="Crime"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
          data-genre="Drama"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
          data-genre="Adventure"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Adventure
          </a>
        </li>
        <li
          className="catalog__genres-item "
          data-genre="Comedy"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Bronson
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Pulp Fiction
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Midnight Special
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Matrix
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              Legend
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Gangs of new york
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              A Star Is Born
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              style={
                Object {
                  "height": "100%",
                  "objectFit": "cover",
                  "verticalAlign": "top",
                  "width": "100%",
                }
              }
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/8"
              onClick={[Function]}
            >
              Beach
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`App render without store renders App component's preloader (no store here) 1`] = `
<div>
  Loading...
</div>
`;
